cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestBase.java

```diff
@@ -156,9 +157,26 @@ public Boolean call() throws Exception {
         if (!getMonitorClient().allSystemServicesOk()) {
           return false;
         }
-        // Check that the dataset service is up with list(). If list() does not throw exception, which means the http
-        // request receives response status HTTP_OK and dataset service is up, return true.
-        getNamespaceClient().list();
+
+        // Wait for the default namespace to exist otherwise there can be a race condition where namespace is being
+        // created but the integration test has started
+        Tasks.waitFor(true, new Callable<Boolean>() {
```

Comment:
no need to wrap this in a Tasks#waitFor.
This method itself will be executed in a Tasks#waitFor (look at the usages of this method).